billy-portugal-ebean/src/main/java/com/premiumminds/billy/portugal/persistence/dao/ebean/DAOPTCreditNoteImpl.java

```diff
@@ -41,38 +48,54 @@ public PTCreditNoteEntity getEntityInstance() {
     }
 
     @Override
-    public List<PTCreditNoteEntity> getBusinessCreditNotesForSAFTPT(UID uid, Date from, Date to) {
-        /*QJPAPTCreditNoteEntity creditNote = QJPAPTCreditNoteEntity.jPAPTCreditNoteEntity;
-        
-        JPAQuery query = this.createQuery();
-        
-        query.from(creditNote)
-                .where(creditNote.instanceOf(JPAPTCreditNoteEntity.class).and(creditNote.date.between(from, to))
-                        .and(this.toDSL(creditNote.business, QJPAPTBusinessEntity.class).uid.eq(uid.toString())));
-        
-        List<PTCreditNoteEntity> result = this.checkEntityList(query.list(creditNote), PTCreditNoteEntity.class);
-        return result;*/
-        return null;
+    public List<PTCreditNoteEntity> getBusinessCreditNotesForSAFTPT(UID businessUid, Date from, Date to) {
+        List<JPAGenericInvoiceEntity> invoices =
+                this.queryInvoice().business.uid.eq(businessUid.toString()).date.between(from, to).findList();
+
+        ArrayList<PTCreditNoteEntity> creditNotes = new ArrayList<>();
+        for (JPAGenericInvoiceEntity invoice : invoices) {
+            if (invoice instanceof JPAPTCreditNoteEntity) {
+                creditNotes.add((JPAPTCreditNoteEntity) invoice);
+            }
+        }
+        return creditNotes;
     }
 
     @SuppressWarnings("unchecked")
     @Override
     public List<PTCreditNote> findByReferencedDocument(UID uidCompany, UID uidInvoice) {
-        /*QJPAPTCreditNoteEntity creditNote = QJPAPTCreditNoteEntity.jPAPTCreditNoteEntity;
-        QJPAPTCreditNoteEntryEntity entry = QJPAPTCreditNoteEntryEntity.jPAPTCreditNoteEntryEntity;
-        QJPAPTGenericInvoiceEntity invoice = QJPAPTGenericInvoiceEntity.jPAPTGenericInvoiceEntity;
+        JPAGenericInvoiceEntity invoice = this.queryInvoice().uid.eq(uidInvoice.toString()).findOne();
```

Comment:
The previous code did not check for nulls because it (incorrectly) tried to obtain a list of invoices from this query, which can actually only return 1 object max. Now that the query only obtains a single object.
Before, finding no results would lead to an empty list being returned. I've changed the new code to do the same; return an empty list if no result is found (the invoice is null).